exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/AbstractParquetGroupScan.java

```diff
@@ -281,31 +265,54 @@ public AbstractGroupScanWithMetadata applyFilter(LogicalExpression filterExpr, U
 
       logger.debug("All row groups have been filtered out. Add back one to get schema from scanner");
 
+      Map<Path, SegmentMetadata> segmentsMap = getNextOrEmpty(getSegmentsMetadata().values()).stream()
+          .collect(Collectors.toMap(SegmentMetadata::getPath, Function.identity()));
+
       Map<Path, FileMetadata> filesMap = getNextOrEmpty(getFilesMetadata().values()).stream()
-          .collect(Collectors.toMap(FileMetadata::getLocation, Function.identity()));
+          .collect(Collectors.toMap(FileMetadata::getPath, Function.identity()));
 
       Multimap<Path, RowGroupMetadata> rowGroupsMap = LinkedListMultimap.create();
-      getNextOrEmpty(getRowGroupsMetadata().values()).forEach(entry -> rowGroupsMap.put(entry.getLocation(), entry));
+      getNextOrEmpty(getRowGroupsMetadata().values()).forEach(entry -> rowGroupsMap.put(entry.getPath(), entry));
 
-      builder.withRowGroups(rowGroupsMap)
+      filteredMetadata.withRowGroups(rowGroupsMap)
           .withTable(getTableMetadata())
+          .withSegments(segmentsMap)
           .withPartitions(getNextOrEmpty(getPartitionsMetadata()))
           .withNonInterestingColumns(getNonInterestingColumnsMetadata())
           .withFiles(filesMap)
           .withMatching(false);
     }
 
-    if (builder.getOverflowLevel() != MetadataLevel.NONE) {
-      logger.warn("applyFilter {} wasn't able to do pruning for  all metadata levels filter condition, since metadata count for " +
-            "{} level exceeds `planner.store.parquet.rowgroup.filter.pushdown.threshold` value.\n" +
-            "But underlying metadata was pruned without filter expression according to the metadata with above level.",
-          ExpressionStringBuilder.toString(filterExpr), builder.getOverflowLevel());
+    if (filteredMetadata.getOverflowLevel() != MetadataType.NONE) {
+      if (logger.isWarnEnabled()) {
```

Comment:
No objections for this change but what are the odds of warn level being disabled? :)